src/mod_mam_sql.erl

```diff
@@ -324,31 +373,65 @@ export(_Server) ->
                 LPeer = jid:encode(jid:tolower(Peer)),
                 XML = fxml:element_to_binary(Pkt),
                 Body = fxml:get_subtag_cdata(Pkt, <<"body">>),
+								Thread = fxml:get_subtag_cdata(Pkt, <<"thread">>),
                 SType = misc:atom_to_binary(Type),
                 SqlType = ejabberd_option:sql_type(Host),
                 case SqlType of
-	                mssql -> [?SQL_INSERT(
-	                    "archive",
-	                    ["username=%(SUser)s",
-	                     "server_host=%(LServer)s",
-	                     "timestamp=%(TStmp)d",
-	                     "peer=%(LPeer)s",
-	                     "bare_peer=%(BarePeer)s",
-	                     "xml=N%(XML)s",
-	                     "txt=N%(Body)s",
-	                     "kind=%(SType)s",
-	                     "nick=%(Nick)s"])];
-	                _ -> [?SQL_INSERT(
-	                    "archive",
-	                    ["username=%(SUser)s",
-	                     "server_host=%(LServer)s",
-	                     "timestamp=%(TStmp)d",
-	                     "peer=%(LPeer)s",
-	                     "bare_peer=%(BarePeer)s",
-	                     "xml=%(XML)s",
-	                     "txt=%(Body)s",
-	                     "kind=%(SType)s",
-	                     "nick=%(Nick)s"])]
+	                mssql ->
+										case Thread of
+											<<"null">> ->
+												[?SQL_INSERT(
+													"archive",
+													["username=%(SUser)s",
+													"server_host=%(LServer)s",
+													"timestamp=%(TStmp)d",
+													"peer=%(LPeer)s",
+													"bare_peer=%(BarePeer)s",
+													"xml=N%(XML)s",
+													"txt=N%(Body)s",
+													"kind=%(SType)s",
+													"nick=%(Nick)s"])];
+											_ ->
+												[?SQL_INSERT(
+													"archive",
+													["username=%(SUser)s",
+													"server_host=%(LServer)s",
+													"timestamp=%(TStmp)d",
+													"peer=%(LPeer)s",
+													"bare_peer=%(BarePeer)s",
+													"xml=N%(XML)s",
+													"txt=N%(Body)s",
+													"kind=%(SType)s",
+													"nick=%(Nick)s"])]
+											end;
+	                _ ->
```

Comment:
This is another flow in which a message could be added to the table, so the same flow was added.